debezium-server-rabbitmq/src/main/java/io/debezium/server/rabbitmq/RabbitmqStreamChangeConsumer.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.server.rabbitmq;
+
+import java.io.IOException;
+import java.time.temporal.ChronoUnit;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.enterprise.context.Dependent;
+import javax.inject.Named;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.rabbitmq.client.AMQP;
+import com.rabbitmq.client.Channel;
+import com.rabbitmq.client.Connection;
+import com.rabbitmq.client.ConnectionFactory;
+
+import io.debezium.DebeziumException;
+import io.debezium.engine.ChangeEvent;
+import io.debezium.engine.DebeziumEngine;
+import io.debezium.engine.DebeziumEngine.RecordCommitter;
+import io.debezium.engine.Header;
+import io.debezium.server.BaseChangeConsumer;
+
+/**
+ * Implementation of the consumer that delivers the messages into RabbitMQ Stream destination.
+ *
+ * @author Olivier Boudet
+ *
+ */
+@Named("rabbitmq")
+@Dependent
+public class RabbitmqStreamChangeConsumer extends BaseChangeConsumer implements DebeziumEngine.ChangeConsumer<ChangeEvent<Object, Object>> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RabbitmqStreamChangeConsumer.class);
+
+    private static final String PROP_PREFIX = "debezium.sink.rabbitmq.";
+    @ConfigProperty(name = PROP_PREFIX + "host")
+    String host;
+
+    @ConfigProperty(name = PROP_PREFIX + "port", defaultValue = "5672")
+    int port;
+
+    @ConfigProperty(name = PROP_PREFIX + "auth.enabled", defaultValue = "true")
+    boolean authEnabled;
+
+    @ConfigProperty(name = PROP_PREFIX + "auth.username", defaultValue = "guest")
+    String username;
+
+    @ConfigProperty(name = PROP_PREFIX + "auth.password", defaultValue = "guest")
+    String password;
+
+    @ConfigProperty(name = PROP_PREFIX + "virtualHost", defaultValue = "/")
+    String virtualHost;
+
+    @ConfigProperty(name = PROP_PREFIX + "routingKey", defaultValue = "")
+    Optional<String> routingKey;
+
+    Connection connection;
+
+    Channel channel;
+
+    @PostConstruct
+    void connect() {
+        ConnectionFactory factory = new ConnectionFactory();
```

Comment:
It seems to me that `ConnectionFactory` supports configuration via properties too. If it is the case then please use the same approach as https://debezium.io/documentation/reference/2.1/operations/debezium-server.html#pulsar-client
It is the most flexible approach and enable user to do arbitrary configuration.

Reply:
I changed the way I initizalize the `ConnectionFactory` to use properties, but I had needed to change  `getConfigSubset` to return a `Map<String, String>` instead of a `Map<String, Object>`. Is that change OK for you ?

Reply:
@olivierboudet If you don't mnd I'd prefer the opposite approach. Keep the common code and do the conversion at this sink level.